master/internal/experiment/bulk_action.go

```diff
@@ -548,16 +581,14 @@ func UnarchiveExperiments(ctx context.Context, system *actor.System,
 		Model(&expChecks).
 		Column("e.archived").
 		Column("e.id").
-		ColumnExpr("e.state IN (?) AS state", bun.In([]string{
-			"CANCELED",
-			"COMPLETED",
-			"ERROR",
-		}))
+		ColumnExpr("e.state IN (?) AS state", bun.In(terminalExperimentStates))
 
 	if filters == nil {
 		query = query.Where("e.id IN (?)", bun.In(experimentIds))
 	} else {
-		query = queryBulkExperiments(query, filters)
+		query = queryBulkExperiments(query, filters).
+			Where("archived").
```

Comment:
same question here, why not just always filter already unarchived (even when filters == nil) and just make this fully idempotent.

Reply:
i get that one is asking for a specific list, and this is doing it by filters, so it makes sense.. but idk

Reply:
this can be decided later, actually, it is fine.

Reply:
though it does kinda feel like attempting to delete an experiment that has checkpoints registered in the model by way of catching it in a filter, then just seeing it not get deleting without and user feedback could cause confusion.

Reply:
said another way, i guess the difference between archived/unarchived and delete is that for archived/unarchived, we are filtering it because the attempted action is already done, and for delete we are filtering it because it cannot be done. and it seems this difference is significant enough to warrant a different approach.

Reply:
ok so after a long ramble, i think for delete we should not filter `versions = 0` and just return the error if there are versions.

Reply:
ok I can remove the versions = 0 query